didman/api/v1/api.go

```diff
@@ -94,31 +94,50 @@ func (w *Wrapper) Routes(router core.EchoRouter) {
 	}))
 }
 
-// AddEndpoint handles calls to add a service. It only checks params and sets the correct return status code.
-// didman.AddEndpoint does the heavy lifting.
-func (w *Wrapper) AddEndpoint(ctx context.Context, request AddEndpointRequestObject) (AddEndpointResponseObject, error) {
-	id, err := did.ParseDID(request.Did)
+func (w *Wrapper) addOrUpdateEndpoint(
+	ctx context.Context, requestDID string, properties EndpointProperties,
+	operation func(ctx context.Context, id did.DID, serviceType string, endpoint url.URL) (*did.Service, error),
+) (*did.Service, error) {
+	id, err := did.ParseDID(requestDID)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(strings.TrimSpace(request.Body.Type)) == 0 {
+	if len(strings.TrimSpace(properties.Type)) == 0 {
 		return nil, core.InvalidInputError("invalid value for type")
 	}
 
-	u, err := url.Parse(request.Body.Endpoint)
+	endpoint, err := url.Parse(properties.Endpoint)
```

Comment:
consider using https://pkg.go.dev/net/url#ParseRequestURI which is more strict.

Reply:
Dangerously out of scope change (changing existing behavior)